### ISSUE TITLE
Valid for one year..

### ISSUE THREAD

**Issue by tbone2k-git** [weolar/xpchrome#11]
Hello! o)

What does the text mean at the release/download?
"run on windows xp sp3, valid for one year"

I would like to know, is there some certificate which will expire or why would the executable not run after 1 year?

Thank you for doing what you do! o)

**Comment by andika207** [comment]
> What does the text mean at the release/download? "run on windows xp sp3, valid for one year"

![screenshot 3](https://github.com/weolar/xpchrome/assets/47242117/c77b6f9f-d415-4c01-bf57-fe92b371b72a)

**Comment by win32ss** [comment]
Set your system date to July 31, 2024 or later, and the browser will crash by trying to write to 0x0 on startup.

**Comment by tbone2k-git** [comment]
Ok, I was expecting some other cause, but still interesting! o)
Is this limitation/bug something which can be addressed in the future?

Thanks! o)

**Comment by rereser** [comment]
https://www.nirsoft.net/utils/run_as_date.html

**Comment by tbone2k-git** [comment]
Some nice guy sent me a link to "RunAsDate" from NirSoft (thank you! o). Some of you probably know the NirSoft tools, some are very handy, always worth checking them out. I'm not sure though, whether faking the system date is a proper workaround to this "one year" issue, but ok, it's some kind of progress! o)

In the meanwhile I tested the zipped chromium release and had some surfing with it, it works. It puts the CPU to 100% when playing a yt video, but I guess that is expected with missing hardware acceleration and/or modern instructions sets not being used(?).

Thanks! o)

**Comment by andika207** [comment]
> Some nice guy sent me a link to "RunAsDate" from NirSoft (thank you! o). 

@rereser :ok_hand:

**Comment by tbone2k-git** [comment]
@andika207 
I did not put the name of the sender here, because he obviously wanted to keep it private.
I don't get what you did here.

**Comment by SaranSDS008** [comment]
This version of chrome for xp seems to have timebomb for one year. Therefore @win32ss provided a way to mitigate this issue:

The following is a patch to remove the XP Chrome 115 August 2024 (and August of every year following due to logic error) timebomb. Install a Hex Editor (like HxD or ImHex or any Hex Editor if your choice), open chrome.dll in it and,
Search for:
83 EC 04 0F B7 84 24 80 00 00 00 3D E8 07 00 00 7C 1F 0F B7 84 24 82 00 00 00 83 F8 08 7C 12 E8

change to:
83 EC 04 0F B7 84 24 80 00 00 00 3D E8 07 00 00 EB 1F 0F B7 84 24 82 00 00 00 83 F8 08 7C 12 E8

**Comment by weolar** [comment]
I would like to explain: Letting Chromium run on XP is a technical solution that I have been researching for several years. Recently, I plan to develop a commercial browser, XPChrome, which was originally intended to be distributed only in China and not intended for overseas distribution. However, I saw that many friends overseas were discussing running Chrome on XP, so I removed some code related to user permission verification and update detection, and allowed everyone to try it out One year. Occasionally, XPChrome still has data connections to Chinese servers because some code may not have been completely removed. If you like this product, I will also create a version of XPChrome that supports XP running with only local verification and no network verification, and has two-core (IE,chromium) support.

**Comment by Vangelis66** [comment]
> and has two-core (**IE**,chromium) support.

This is indeed a feature common in the **Chinese Chrome forks** (like **360EE**), but I can't, in all honesty, understand its necessity when **targeting old Windows OSs** like XP/Vista; on **XP**, **IE8** (and on **Vista**, **IE9**) is totally unable to handle the web of 2023-4, and not only because of **TLS**/certificate issues; its JS+CSS engines are quite obsolete by now, most popular sites won't render/render broken in the IE web engine 😭 ... 

Even when one uses **XPChrome v115** on **Win7/8.1**, to mitigate evil **Google**'s 😡 deprecation of **Chrome** (past v109) there, **IE11**, certainly more "web able" 👍 than its previous versions, has been abandoned by its vendor (**Microsoft**) and soon (if not already) the same will become true of website authors; so, an "IE core" inside XPChrome looks like a moot point to me... Just my 2c, ofc 😜 ...

FWIW, many thanks for your efforts into creating this project ❤️ ...

**Comment by blackcrack** [comment]
to have a independed ChromeBrowser in any case advantageous, if i see also the whestern Peoples where have no support in XP and Win7 from MS anymore and have therewith a correct Browser, also the "normal" crome support for XP and Win7 is imho.. gone what Google have become with is Win7 and XP no more supportet by Google ?! :  
https://support.google.com/chrome/a/answer/7100626?hl

so, because it is in the states.. is Pushing it to Win10 and up, for bring the Win11 and Cloude with Windows 365 together in the near time ... where was hearing at the HighCord before a couples years ... for bring the whole Peoples under Control in a Cloud-Ordering-System(because the savety&security and by side fishing up comercial informations for commerce acting ).. the UEFI and TPM was only the working befor for bring the peoples the "savety and security into the brain of the peoples" and because that bring all american factorys the peoples t the way what's the States whant.. in direction of China .. but more commercial..and Fasilitys where supporting the American law .. (point, point) for that need we a independed chromebased "silence" Browser wheres run still in the whole OrderingsSystem's like WXP and 7 .. because this, also a massive big thank you for your Browser support :) :+1:   
best  
blacky

**Comment by diderius6** [comment]
Hi,
how to enable WebGL 
in this nice browser
Dietmar

**Comment by weolar** [comment]
> Hi, how to enable WebGL in this nice browser Dietmar

Sorry, running WebGL programs on XPChrome will be slower because hardware acceleration is turned off.

**Comment by weolar** [comment]
> > and has two-core (**IE**,chromium) support.
> 
> This is indeed a feature common in the **Chinese Chrome forks** (like **360EE**), but I can't, in all honesty, understand its necessity when **targeting old Windows OSs** like XP/Vista; on **XP**, **IE8** (and on **Vista**, **IE9**) is totally unable to handle the web of 2023-4, and not only because of **TLS**/certificate issues; its JS+CSS engines are quite obsolete by now, most popular sites won't render/render broken in the IE web engine 😭 ...
> 
> Even when one uses **XPChrome v115** on **Win7/8.1**, to mitigate evil **Google**'s 😡 deprecation of **Chrome** (past v109) there, **IE11**, certainly more "web able" 👍 than its previous versions, has been abandoned by its vendor (**Microsoft**) and soon (if not already) the same will become true of website authors; so, an "IE core" inside XPChrome looks like a moot point to me... Just my 2c, ofc 😜 ...
> 
> FWIW, many thanks for your efforts into creating this project ❤️ ...

You are absolutely right. It seems that the IE engine is not very useful anymore. However, I'm not sure if the Flash plugin is useful for you. I also plan to support NPAPI and Flash.

**Comment by Cixert** [comment]
So can we conclude that the 1 year time bomb starts on the version's production date? Or what date does the time bomb start?
